designer/server/src/main/scala/pl/touk/nussknacker/ui/api/DeploymentApiHttpService.scala

```diff
@@ -0,0 +1,81 @@
+import pl.touk.nussknacker.ui.api.description.DeploymentApiEndpoints.Dtos.DeploymentError.{NoPermission, NoScenario}
+import pl.touk.nussknacker.ui.process.ProcessService
+import pl.touk.nussknacker.ui.process.ProcessService.GetScenarioWithDetailsOptions
+import pl.touk.nussknacker.ui.process.deployment.DeploymentService
+import pl.touk.nussknacker.ui.security.api.{AuthenticationResources, LoggedUser}
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+class DeploymentApiHttpService(
+    authenticator: AuthenticationResources,
+    scenarioService: ProcessService,
+    deploymentService: DeploymentService
+)(implicit executionContext: ExecutionContext)
+    extends BaseHttpService(authenticator) {
+
+  private val endpoints = new DeploymentApiEndpoints(authenticator.authenticationMethod())
+
+  expose {
+    endpoints.requestScenarioDeploymentEndpoint
+      .serverSecurityLogic(authorizeKnownUser[DeploymentError])
+      .serverLogicEitherT { implicit loggedUser =>
+        // TODO (next PRs): use params
+        { case (scenarioName, deploymentId, request) =>
+          for {
+            scenarioDetails <- getScenarioDetailsByName(scenarioName)
+            _ <- EitherT.fromEither[Future](
+              Either.cond(loggedUser.can(scenarioDetails.processCategory, Permission.Deploy), (), NoPermission)
+            )
+            // TODO (next PRs): Currently it is done sync, but eventually we should make it async and add an endpoint for deployment status verification
+            _ <- extractErrors(
+              deploymentService
+                .deployProcessAsync(
+                  scenarioDetails.idWithNameUnsafe,
+                  savepointPath = None,
+                  comment = None
+                )
+                .flatten
+            )
+          } yield ()
+        }
+      }
+  }
+
+  // TODO: It is a copy-paste of akka-based AuthorizeProcess variant. We should:
+  //       - reuse fetched scenario inside DeploymentService.deployProcessAsync
+  //       - extract this boilerplate for permission checking to some class - but before that we should rethink errors
+  //         type hierarchy. currently we have dedicated errors for each endpoint which makes this hard to achieve
+  private def getScenarioDetailsByName(
+      scenarioName: ProcessName
+  )(implicit loggedUser: LoggedUser): EitherT[Future, DeploymentError, ScenarioWithDetails] =
+    for {
+      scenarioId <- EitherT.fromOptionF(scenarioService.getProcessId(scenarioName), NoScenario(scenarioName))
+      scenarioDetails <- extractErrors(
+        scenarioService.getLatestProcessWithDetails(
+          ProcessIdWithName(scenarioId, scenarioName),
+          GetScenarioWithDetailsOptions.detailsOnly
+        )
+      )
+    } yield scenarioDetails
+
+  private def extractErrors[T](future: Future[T]): EitherT[Future, DeploymentError, T] = {
```

Comment:
I like the `eitherifyBusinessError`. I changed to it. But TBH it is still tricky because:
1. It doesn't eitherify all business errors, only `UnauthorizedError`, but it may change in the future
2. This `UnauthorizedError` is handled by `CustomAuthorizationError` which is a hacky business error that is translated into the security error in `BaseHttpService` - security error is

Reply:
For purposes of this PR imo its good. The name can be shortened as just `eitherify` or `eitherifyErrors` - I'll leave it up to you - all sounds good enough to me.